Lib/cProfile.py

```diff
@@ -140,16 +144,23 @@ def main():
     sys.argv[:] = args
 
     if len(args) > 0:
-        progname = args[0]
-        sys.path.insert(0, os.path.dirname(progname))
-        with open(progname, 'rb') as fp:
-            code = compile(fp.read(), progname, 'exec')
-        globs = {
-            '__file__': progname,
-            '__name__': '__main__',
-            '__package__': None,
-            '__cached__': None,
-        }
+        if options.module:
+            code = "runpy.run_module(modname, run_name='__main__')"
+            globs = {
+                'runpy': runpy,
+                'modname': args[0]
```

Comment:
Sure, I'll fix that and also add `NEWS` entry :)

Reply:
Hey @ncoghlan !

So, one of the thing I tried from your code snippet was this:

```
        if options.module:
            modname = args[0]
            code = f"run_module({modname}, run_name='__main__')"
            globs = {
                'run_module': runpy.run_module,
                'modname': modname
            }
```

and if I run `./python.exe -m cProfile -m timeit -n 1` to test it, it throws `NameError` for `timeit`. But this is not the case without f-strings. I'm unable to understand this behaviour. Can you please help?

Reply:
This is not the case if I avoid using f-string and provide the context of running the module as globals to the `runctx` method. So, I'm not sure, what is wrong. Help would be much appreciated.

Reply:
I think `modname` would need to be fed through repr(), e.g. `f"run_module({modname!r}, run_name='__main__')"`.
That said, I think your original code is fine and I don't find @ncoghlan's suggestion more readable :-)

Reply:
@pitrou is right, and the missing `!r` was just a bug in my suggested code. However, there's merit in the idea of using just one mechanism for passing values in (the `globs` dict) rather than two (string interpolation and the globs dict).

So while I still think it makes sense to pass in a reference to the `run_module` itself, rather than one to the `runpy` module, I withdraw the suggestion of using string interpolation to pass in the module name to run.